objects/src/accounts/storage/header.rs

```diff
@@ -0,0 +1,109 @@
+use alloc::vec::Vec;
+
+use vm_core::utils::{ByteReader, ByteWriter, Deserializable, Serializable};
+use vm_processor::DeserializationError;
+
+use super::{AccountStorage, StorageSlotType, Word};
+
+// ACCOUNT STORAGE HEADER
+// ================================================================================================
+
+/// Account storage header is a lighter version of the [AccountStorage] storing
+/// only the [StorageSlotType] and associated values of the [super::StorageSlot]s
+/// contained in the storage.
+///
+/// The use of a header is useful in the situation where the storage is heavy (i.g. multiple Mb's),
+/// and should be enough to execute most transactions.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AccountStorageHeader {
+    slots: Vec<(StorageSlotType, Word)>,
+}
+
+impl AccountStorageHeader {
```

Comment:
I would probably add a couple more methods here - like:
```rust
pub fn num_slots(&self) -> usize

pub fn slot(&self, index: usize) -> &(StorageSlotType, Word)
```

Reply:
Agreed, these methods could be useful. 

I had previously added, then removed methods like: 

```rust
pub fn types(&self)
pub fn values(&self)
```

The compiler was shouting at me for unused code hence I did not know what was the way to go in the specific case of Miden, should I `allow dead code` these methods or should I only leave used methods?

Reply:
We should export `AccountStorageHeader` publicly from the crate (this will also fix these warnings).